### PR TITLE
Updates to the build recipe for ISIS releases for the 3.7.0 release candidate.

### PR DESCRIPTION
**Pull request by SgStapleton**
Just small changes for release candidate build.

## Description

Updating recipes and version files for the conda build process.

## Related Issue

Release 3.7.0-RC

## Motivation and Context

Required for the release process

## How Has This Been Tested?

(None needed)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,12 +12,12 @@
 #       A Public Release for ISIS3.6.1:                        {% set version = "3.6.1" %}
 #       A Release Candidate for ISIS3.6.1:                     {% set version = "3.6.1_RC" %}
 #       A custom build of ISIS3.6.1 for the CaSSIS mission:    {% set version = "3.6.1_cassis" %}
-{% set version = "3.6.1" %}
+{% set version = "3.7.0_RC" %}
```
Comment (jlaura) [inline recipe/meta.yaml:16]: I am just noting what you told me today - while semantic versioning requires hyphenation, we have to use an underscore in the meta.yml. :+1:

File: recipe/meta.yaml [hunk 2]
```diff
@@ -12,12 +12,12 @@
...
 
 # This is the branch that conda build will pull from the ISIS repository. If you are building for a
 # general public release, leave the branch as "release". If you are building a custom build, like
 # for a mission for example, simply change the value to the branch of choice.
-{% set git_branch = "release" %}
+{% set git_branch = "dev" %}
```
Comment (jlaura) [inline recipe/meta.yaml:20]: Should this be a branch or should this be a tag?

Comment (jessemapel) [reply recipe/meta.yaml:20]: We should be doing the build off of a release tarball

Comment (SgStapleton) [reply recipe/meta.yaml:20]: I guess I'm confused about the question here...?

Comment (krlberry) [reply recipe/meta.yaml:20]: I'm even more confused. I can't even find a reference to `git_branch` in the docs https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-section.

But it also looks like `git_tag` exists, in which case we could use this 3.7.0_RC tag to do the build, which seems correct to me (and is probably what @jlaura was suggesting)? https://stackoverflow.com/questions/22546878/possible-to-do-conda-build-from-github-branch

Comment (jessemapel) [reply recipe/meta.yaml:20]: @SgStapleton @kberryUSGS For the RCs, using the tag makes sense. For the full 3.7.0 release we should be creating a release tarball on github. Then, for the release build we should be using that tarball.

See how we build [the csm on conda-forge](https://github.com/conda-forge/csm-feedstock/blob/master/recipe/meta.yaml#L8).

**Comment by blandoplanet** [comment]
Will 3.7.0_RC be available internally?

**Comment by krlberry** [comment]
@blandoplanet's question was answered on [astrodiscuss](https://astrodiscuss.usgs.gov/t/a-release-candidate-for-isis-3-7-0-is-now-available/147)

**Comment by krlberry** [comment]
@jlaura @jessemapel Can this get merged as-is since it is actually what was done to make the build?
I'm not sure where this information should go, but I made a note to myself to use something like:
`{% set git_tag = "3.7.0_RC2" %}` for the next RC.

**Review by krlberry — APPROVED** [review]
:+1: I say we make a note of @jessemapel's suggestion and merge this as a record of what was done to build the first RC.

**Comment by jessemapel** [comment]
@kberryUSGS I have no problem merging this. We should eventually go to building off of tarballs.

File: recipe/meta.yaml (hunk 2 above)
Comment (SgStapleton) [reply recipe/meta.yaml:20]: I guess the main concern I have here is when there needs to be a special build for a particular mission. They will need to either pull from a particular branch, or they will need to create a tag for their build in the repository. The latter would probably be preferred to keep track of the snapshots of the repo at particular builds, but I just wanted to make sure this makes sense to everyone.

**Pull request merged**